src/test/run-pass/extern-types-pointer-cast.rs

```diff
@@ -0,0 +1,30 @@
+// Copyright 2017 The Rust Project Developers. See the COPYRIGHT
```

Comment:
It would be better to use a pointer with an "interesting" bit pattern rather than `0`, to make sure we are not truncating anything.

Reply:
Done. I've had to use `cfg(target_pointer_width)` to make sure all bits have something interesting.